dipy/nn/finta_model.py

```diff
@@ -0,0 +1,435 @@
+
+        Parameters
+        ----------
+        input_data : tf.Tensor or np.ndarray
+            Data to run through the decoder.
+            Should be of the shape of the latent
+            space: (n_samples, latent_space_dims).
+
+        Returns
+        -------
+        tf.Tensor
+            Decoded input in the original
+            data space, with the shape
+            (n_samples, 256, 3).
+        """
+        z = input_data
+        fc = self.fc2(z)
+
+        # Reshape to match encoder output size
+        fc_reshape = tf.reshape(fc, (-1, self.encoder_out_size[0],
+                                     self.encoder_out_size[1]))
+
+        h1 = tf.nn.relu(self.decod_conv1(fc_reshape))
+        h2 = self.upsampl1(h1)
+        h3 = tf.nn.relu(self.decod_conv2(h2))
+        h4 = self.upsampl2(h3)
+        h5 = tf.nn.relu(self.decod_conv3(h4))
+        h6 = self.upsampl3(h5)
+        h7 = tf.nn.relu(self.decod_conv4(h6))
+        h8 = self.upsampl4(h7)
+        h9 = tf.nn.relu(self.decod_conv5(h8))
+        h10 = self.upsampl5(h9)
+        h11 = self.decod_conv6(h10)
+
+        return h11
+
+
+def init_model(latent_space_dims=32, kernel_size=3):
+    r"""
+    Initialize the model with the given latent
+    space dimensions and kernel size.
+    Instantiates the Encoder, the Decoder, and
+    returns the model comprising both blocks.
+
+    Parameters
+    ----------
+    latent_space_dims : int, optional
+        Number of dimensions of the latent
+        space where the data will be encoded
+        Default: 32
+    kernel_size : int, optional
+        Length of the 1D kernel used in the
+        convolutional layers of the Encoder
+        and the Decoder
+        Default: 3
+
+    Returns
+    -------
+    keras.Model
+        Encoder model
+    keras.Model
+        Decoder model
+    tuple
+        Size of the convolutional output of
+        the Encoder block.
+    """
+    input_data = keras.Input(shape=(256, 3), name='input_streamline')
+
+    # encode
+    encoder = Encoder(latent_space_dims=latent_space_dims,
+                      kernel_size=kernel_size)
+    encoded = encoder(input_data)
+    # Instantiate encoder model
+    model_encoder = Model(input_data, encoded, name="Encoder")
+
+    # decode
+    decoder = Decoder(encoder.encoder_out_size,
+                      kernel_size=kernel_size)
+    decoded = decoder(encoded)
+    output_data = decoded
+    # Instantiate decoder model
+    model_decoder = Model(encoded, decoded, name="Decoder")
+
+    # Instantiate model and name it
+    return model_encoder, model_decoder, encoder.encoder_out_size
+
+
+class IncrFeatStridedConvFCUpsampReflectPadAE(Model):
+    """
+    Strided convolution-upsampling-based
+    AutoEncoder using reflection-padding and
+    increasing feature maps in decoder.
+    Architecture based on Jon Haitz Legarreta Gorroño's
+    PyTorch implementation, from his FINTA [1] work.
+    [1] https://doi.org/10.1016/j.media.2021.102126
```

Comment:
You can save mentioning my name here. We have adopted `sphinxcontrib-bibtex` to increase consistency across citations and to avoid Sphinx raising +1000 warnings. You can have a look at the open PRs to see how to change this.

Add some more explanation about the AE.